zanata-war/src/main/java/org/zanata/adapter/XliffAdapter.java

```diff
@@ -0,0 +1,147 @@
+            IllegalArgumentException {
+
+        if (sourceLocale == null) {
+            throw new IllegalArgumentException("Source locale cannot be null");
+        }
+
+        if (fileUri == null) {
+            throw new IllegalArgumentException("Document URI cannot be null");
+        }
+
+        XliffReader xliffReader = new XliffReader();
+
+        File tempFile = null;
+        Resource doc = null;
+        try {
+            tempFile = new File(fileUri);
+            doc = xliffReader.extractTemplate(tempFile, sourceLocale,
+                            tempFile.getName(),
+                            XliffCommon.ValidationType.CONTENT.name());
+        } catch (IOException e) {
+            throw new FileFormatAdapterException(
+                    "Could not open the URL. The URL is OK but the input stream could not be opened.\n"
+                            + e.getMessage(), e);
+        }
+        return doc;
+    }
+
+    @Override
+    public TranslationsResource parseTranslationFile(URI fileUri,
+            String localeId, Optional<String> params)
+            throws FileFormatAdapterException, IllegalArgumentException {
+
+        XliffReader xliffReader = new XliffReader();
+        TranslationsResource targetDoc = null;
+        File transFile = null;
+
+        try {
+            transFile = new File(fileUri);
+            targetDoc = xliffReader.extractTarget(transFile);
+        } catch (FileNotFoundException e) {
+            throw new FileFormatAdapterException(
+                "Could not open the URL. The URL is OK but the input stream could not be opened.\n"
+                    + e.getMessage(), e);
+        } finally {
+            if (transFile != null) {
+                if (!transFile.delete()) {
+                    log.warn(
+                        "unable to remove temporary file {}, marked for delete on exit",
+                        transFile.getAbsolutePath()); transFile.deleteOnExit();
+                }
+            }
+        }
+        return targetDoc;
+    }
+
+    @Override
+    public void writeTranslatedFile(OutputStream output, URI originalFile,
+            Resource resource, TranslationsResource translationsResource,
+            String locale, Optional<String> params)
+            throws FileFormatAdapterException, IllegalArgumentException {
+
+        //write source string with empty translation
+        boolean createSkeletons = true;
+
+        File tempFile = null;
+        try {
+            tempFile = File.createTempFile("filename", "extension");
+            XliffWriter.writeFile(tempFile, resource, locale,
+                translationsResource, createSkeletons);
+
+            byte[] buffer = new byte[4096]; // To hold file contents
+            int bytesRead;
+            FileInputStream input = new FileInputStream(tempFile);
+            while ((bytesRead = input.read(buffer)) != -1) {
+                output.write(buffer, 0, bytesRead);
+            }
```

Comment:
- [ ] replace this with a call to the method `writeFileOutputToStream(...)` that I mentioned earlier.